opencog/atoms/base/Atom.cc

```diff
@@ -227,6 +228,21 @@ void Atom::markForRemoval(void)
     _flags |= MARKED_FOR_REMOVAL;
 }
 
+bool Atom::isChecked() const
+{
+    return (_flags & CHECKED) != 0;
+}
+
+void Atom::setChecked(void)
+{
+    _flags |= CHECKED;
+}
+
+void Atom::setUnchecked(void)
+{
+    _flags &= ~CHECKED;
+}
+
```

Comment:
Not having a way to mark the atoms means you'd have to search for atoms in both directions for AtomSpace equality checking. For atomspaces A and B you'd have to make sure each atom in A was in B, and then check to see that each atom in B was also in A. By setting a flag, you only need to make sure that every atom in B has the flag set when you are done.

I thought a general-purpose marking method would be useful for other test generation cases. And we already have `_flags` being used for removal, and it looked like other temporary uses in the past. So it doesn't add storage space to Atoms.

I can certainly take this out if you really don't like it.